src/operator.rs

```diff
@@ -42,14 +41,23 @@ impl<'mir, 'tcx> EvalContextExt<'tcx> for super::MiriEvalContext<'mir, 'tcx> {
             Eq | Ne => {
                 // This supports fat pointers.
                 let eq = match (*left, *right) {
-                    (Immediate::Scalar(left), Immediate::Scalar(right)) =>
-                        self.ptr_eq(left.not_undef()?, right.not_undef()?)?,
-                    (Immediate::ScalarPair(left1, left2), Immediate::ScalarPair(right1, right2)) =>
-                        self.ptr_eq(left1.not_undef()?, right1.not_undef()?)? &&
-                        self.ptr_eq(left2.not_undef()?, right2.not_undef()?)?,
+                    (Immediate::Scalar(left), Immediate::Scalar(right)) => {
+                        self.ptr_eq(left.not_undef()?, right.not_undef()?)?
+                    }
+                    (
+                        Immediate::ScalarPair(left1, left2),
+                        Immediate::ScalarPair(right1, right2),
+                    ) => {
```

Comment:
Breaking the pattern here across multiple lines makes the code basically unreadable IMO.

Reply:
I am unable to find a rustfmt option (https://rust-lang.github.io/rustfmt/) that does this differently

Reply:
Well then we'll have to manually edit the code.

I'd rather have a readable codebase than one where rustfmt is idempotent. Or in other words, when rustfmt does something silly, I won't let it dictate our style (unless you agree with rustfmt).

AFAIK there's some attribute to locally opt-out of rustfmt?

Reply:
yes, I'll do that